### ISSUE TITLE
How to develop with GTK4

### ISSUE THREAD

**Issue by julihermes** [Aylur/astal#64]
**Have you read through the documentation?**
- [x] [Astal docs](https://aylur.github.io/astal/)
- [x] [Library references](https://aylur.github.io/astal/guide/libraries/references)
- [x] [FAQ](https://aylur.github.io/astal/guide/ags/faq)

**Describe what have you tried so far**
Is there a way to test the GTK4 version of Astal?

**Describe your question**
I have did confronting some color theme bugs, Astal3 not see the GTK theme that I using, and I thing is because the GTK3 limitations, so, I think GTK4 will work well.
How to test Astal4? Or is not developed sufficient to this?

**Comment by kotontrion** [comment]
Porting astal to Gtk4 is planned and some preparations for it are already done, but gtk4 is not available as of now.

However you can still use most of the libs with gtk4, but not the core astal lib and the tray lib, as they depend on gtk3. If you want to use js/ts I do not recommend this, as the astal core lib is really useful there. But if you are interested in using gtk4 now, you can look at [my bar](https://github.com/kotontrion/kompass) which uses gtk4 and is written vala.

What issues are you experiencing exactly? I'm sure they can be solved using gtk3.

**Comment by julihermes** [comment]
I am a develop and I start to try make a bar and system settings with Astal, so, I start with the documentation making my first widget, but, the color not respect the theme, and, I want to make my project compatible with the GTK theme used in system.
I using a Catppuccin gtk theme, the dark version, but my widget still using the adwaita light theme.
This theme is a fork of Colloid GTK theme, and, searching for a solution, I found this: https://github.com/vinceliuice/Colloid-gtk-theme/issues/195. Maybe I am wrong, but I figure out that some theme is not fully compatible with GTK3.
Because of this I create this question.

**Comment by Aylur** [comment]
Just had a quick look, they don't seem to expose their colors for gtk3, so your best bet is to just copy their color definitions and use those, or use gtk4 since those colors are exported as css variables.

**Comment by julihermes** [comment]
Ok, I will use your bar as a refer to try make my own.
Thanks!
